src/bun.js/bindings/TTYHelper.cpp

```diff
@@ -0,0 +1,222 @@
+#include <termios.h>
+#include <unistd.h>
+#include <cstdio>
+
+#include "TTYHelper.h"
+
+namespace Zig {
+
+static bool tty__orig_set = false;
```

Comment:
This will break when we have multiple instances of Bun's runtime 

Sort of need this to be part of JSTTYHelper as an object with the relevant fields